docs/decisions/0003-django-setting-toggles.rst

```diff
@@ -0,0 +1,66 @@
+Django Setting Toggles
+======================
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+Django Settings are implicitly defined when they are used.  For example, in ``edx-platform`` the ``FEATURES`` setting dict is used, and an example usage might look like::
+
+    settings.FEATURES.get('EXAMPLE_FEATURE', False)
```

Comment:
Are we talking only about the `FEATURES` settings? Usage of the `FEATURES` setting is definitely the most problematic, because it's a large dict that is used in many different places. But there are other settings which might be (incorrectly) used as follows:

    getattr(settings, "SOMESETTING", "defaultvalue)

It's a different usage, but the same problematic pattern.

Reply:
No. This ADR s scoped to toggles, meaning settings with a True or False value, but to all of them. I can add a non-FEATURES example as well.